vm/systemSmartContracts/auction.go

```diff
@@ -117,11 +117,106 @@ func (s *stakingAuctionSC) Execute(args *vmcommon.ContractCallInput) vmcommon.Re
 		return s.get(args)
 	case "setConfig":
 		return s.setConfig(args)
+	case "changeRewardAddress":
+		return s.changeRewardAddress(args)
+	case "changeValidatorKeys":
+		return s.changeValidatorKeys(args)
+	case "unJail":
+		return s.unJail(args)
 	}
 
 	return vmcommon.UserError
 }
 
+func (s *stakingAuctionSC) unJail(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	return vmcommon.Ok
+}
+
+func (s *stakingAuctionSC) changeRewardAddress(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if len(args.Arguments) < 1 {
+		return vmcommon.UserError
+	}
+	if len(args.Arguments[0]) != len(args.CallerAddr) {
+		return vmcommon.UserError
+	}
+
+	registrationData, err := s.getOrCreateRegistrationData(args.CallerAddr)
+	if err != nil {
+		return vmcommon.UserError
+	}
+
+	if len(registrationData.RewardAddress) == 0 {
```

Comment:
please use only one way for checking the length. the first check in the method checks that the length is less than one and here you check that the length is equal to 0.

Reply:
checked.